keps/sig-auth/20190730-oidc-discovery.md

```diff
@@ -0,0 +1,101 @@
+---
+title: Service Account signing key retrieval
+authors:
+  - "@mikedanese"
+  - "@cceckman"
+owning-sig: sig-auth
+participating-sigs:
+  - sig-auth
+reviewers:
+  - "@liggitt"
+  - "@enj"
+  - "@micahhausler"
+  - "@ericchiang"
+approvers:
+  - "@liggitt"
+  - "@enj"
+  - "@micahhausler"
+  - "@ericchiang"
+editor: TBD
+creation-date: 2018-06-26
+last-updated: 2019-07-30
+status: provisional
+replaces:
+  - "https://github.com/kubernetes/community/pull/2314/"
+---
+
+# Service Account signing key retrieval
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Implementation History](#implementation-history)
+<!-- /toc -->
+
+## Summary
+
+A non-resource API will be added to the apiserver that will expose an [OIDC
+discovery](https://openid.net/specs/openid-connect-discovery-1_0.html)
+compatible configuration. This will allow clients to retrieve a list of
+currently valid signing keys as an array of JWKs. For example, if the service
+account issuer is configured with the name `https://dev.cluster.internal`, the
+apiserver could expose the following configuration:
+
+ ```
+> GET /.well-known/openid-configuration
+{
+  "issuer": "https://dev.cluster.internal",
+  "jwks_uri": "https://dev.cluster.internal/serviceaccountkeys/v1",
+  "authorization_endpoint": "urn:kubernetes:programmatic_authorization",
```

Comment:
I'd move this to grant type, a la:

```
  "grant_types": [
    "urn:kubernetes:grant_type:programmatic_authorization"
  ]
```

And skip the `authorization_endpoint` for now. It would be good to see what complains (or breaks) in an alpha or beta of this feature.

Reply:
Done.